components/org.wso2.finance.open.banking.conformance.mgt/src/main/java/org/wso2/finance/open/banking/conformance/mgt/dao/impl/TestPlanDAOImpl.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.finance.open.banking.conformance.mgt.dao.impl;
+
+import com.google.gson.Gson;
+import org.wso2.finance.open.banking.conformance.mgt.dao.ReportDAO;
+import org.wso2.finance.open.banking.conformance.mgt.dao.TestPlanDAO;
+import org.wso2.finance.open.banking.conformance.mgt.db.DBConnector;
+import org.wso2.finance.open.banking.conformance.mgt.db.SQLConstants;
+import org.wso2.finance.open.banking.conformance.mgt.dto.TestPlanDTO;
+import org.wso2.finance.open.banking.conformance.mgt.models.Report;
+import org.wso2.finance.open.banking.conformance.mgt.testconfig.TestPlan;
+
+import java.sql.*;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Implementation of the TestPlanDAO
+ * Responsible for the database operations regarding test-plans
+ */
```

Comment:
In general cyclomatic complexity of the methods seems high and please check if methods can be further broken down and make them simple.

Reply:
done